src/Security/Security/Cmdlets/AllowedConnections/GetAllowedConnections.cs

```diff
@@ -0,0 +1,71 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------
+using System.Linq;
+using System.Management.Automation;
+using Commands.Security;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Security.Common;
+using Microsoft.Azure.Commands.Security.Models.AllowedConnections;
+using Microsoft.Azure.Commands.SecurityCenter.Common;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Rest.Azure;
+
+namespace Microsoft.Azure.Commands.Security.Cmdlets.AllowedConnections
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AllowedConnections", DefaultParameterSetName = ParameterSetNames.SubscriptionScope), OutputType(typeof(PSSecurityAllowedConnections))]
+    public class AllowedConnections : SecurityCenterCmdletBase
+    {
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceGroupLevelResource, Mandatory = true, HelpMessage = ParameterHelpMessages.ResourceGroupName)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceGroupLevelResource, Mandatory = true, HelpMessage = ParameterHelpMessages.ResourceName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceGroupLevelResource, Mandatory = true, HelpMessage = ParameterHelpMessages.Location)]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.Security/allowedConnections")]
+        public string Location { get; set; }
```

Comment:
the location seems like was retrieved from client not parameter, so this might not be needed

Reply:
The user may be required to provide it in some instances.

Reply:
there are not entries in SDK for location, how this parameter can be applied to the request?
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/securitycenter/Microsoft.Azure.Management.SecurityCenter/src/Generated/AllowedConnectionsOperations.cs

Reply:
The location is required, a user may provide it manually. 
https://docs.microsoft.com/en-us/rest/api/securitycenter/allowedconnections/listbyhomeregion

Reply:
can you point me how to pass the location parameter to SDK code?

Reply:
the method to list by location is [ListByHomeRegionWithHttpMessagesAsync](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/securitycenter/Microsoft.Azure.Management.SecurityCenter/src/Generated/AllowedConnectionsOperations.cs#L262), you never call this method in your cmdlet, what is this -Location parameter for?

if you would please take a look at the SDK code, the ascLocation was retrieved from SecurityCenterClient which was never been initialized, that means either dotnet SDK generation or swagger spec is incorrect.

Please let me know if there are anything wrong with my understanding.

Reply:
In C# SDK "Location" is provided at the client level, in REST the user is required to provide it manually in some cases.

Reply:
when customer provided "-Location", which line will be executed?
[line 51](https://github.com/Azure/azure-powershell/blob/4c92e114d23333c04ea6e807dc10199172c514eb/src/Security/Security/Cmdlets/AllowedConnections/GetAllowedConnections.cs#L51)
[line 57](https://github.com/Azure/azure-powershell/blob/4c92e114d23333c04ea6e807dc10199172c514eb/src/Security/Security/Cmdlets/AllowedConnections/GetAllowedConnections.cs#L57)
or [line 63](https://github.com/Azure/azure-powershell/blob/4c92e114d23333c04ea6e807dc10199172c514eb/src/Security/Security/Cmdlets/AllowedConnections/GetAllowedConnections.cs#L63)?

Based on your logic it was supposed to be line 57 but [ListByHomeRegionWithHttpMessagesAsync](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/securitycenter/Microsoft.Azure.Management.SecurityCenter/src/Generated/AllowedConnectionsOperations.cs#L262) was not been referenced, how can this "-Location" user provided be applied to a REST call?

Reply:
@VeryEarly, thank you, fixed accordingly.